pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -558,6 +555,8 @@ func (c *Controller) GetProxyServiceInstances(proxy *model.Proxy) ([]*model.Serv
 				return out, nil
 			}
 
+			// 1.1. Headless service without selector
+			out = c.endpoints.GetProxyServiceInstances(c, proxy)
```

Comment:
@hzxuzhonghu should n't we return from here?

Reply:
Good catch,  it is not right. I will fix it

Reply:
Ok. Thank you, Can you also please change the comments to explain the flow little better, if possible?